src/PythonMigrationViewExtension/GraphPythonDependencies.cs

```diff
@@ -81,13 +90,17 @@ internal IEnumerable<INodeLibraryDependencyInfo> AddPythonPackageDependency()
             return new[] { packageDependencyInfo };
         }
 
-        // This function returns true, if any of the custom nodes in the input list has an IronPython dependency. 
-        // It traverses all CN's in the given list of customNodes and marks them as true in CustomNodePythonDependency, 
-        // if the prarent custom node or any of its child custom nodes contain an IronPython dependency.
+        /// <summary>
+        /// This recursive function returns true if any of the custom nodes in the input list has an IronPython dependency.
```

Comment:
this seems to be a big change in behavior? Is the diff wrong? Why these changes?

Reply:
for future reference :
found it to be doing unnecessary cycles (recursively too!) to just say `yay` or `nay` at the end, but as discussed, the full dependency map dictionary is actually needed for displaying the affected nodes in the warning modal.

Reply:
I'll revert behaviour to how it was

Reply:
- `IsIronPythonPackageLoaded` - logic is the same
- `ContainsIronPythonDependencyInCurrentWS` (now `CurrentWorkspaceHasIronPythonDependency`) - reverted to original logic
-